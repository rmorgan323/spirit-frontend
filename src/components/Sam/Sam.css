/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.Sam .sam-image-holder {
  width: 750px;
  margin: 100px auto;
  display: block;
  position: relative; }

.Sam .sam-session-dots {
  background-color: #BFAFD0;
  width: 18px;
  height: 18px;
  padding: 8px 6px 12px 14px;
  margin: 10px;
  border-radius: 100px;
  border: 6px solid #BFAFD0;
  position: absolute;
  transition: all .2s ease-in-out;
  cursor: pointer; }

.Sam .sam-session-dots:hover {
  border: 6px solid #7F60A4; }

.Sam p {
  margin: 0;
  font-size: 18px;
  color: #EAE5F0; }

.Sam .session-dots-1 {
  top: 280px;
  left: 5px; }

.Sam .session-dots-2 {
  top: 283px;
  left: 248px; }

.Sam .session-dots-3 {
  top: 283px;
  left: 498px; }

.Sam .session-dots-4 {
  top: 130px;
  left: 149px; }

.Sam .session-dots-5 {
  top: 125px;
  left: 325px; }

.Sam .session-dots-6 {
  top: 129px;
  left: 679px; }

.Sam .fair-image-holder {
  width: 525px;
  display: block;
  margin: 0 auto 50px; }
