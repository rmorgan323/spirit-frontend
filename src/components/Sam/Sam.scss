@import '../../variables.scss';

.Sam {

  .sam-image-holder {
    width: 750px;
    margin: 100px auto;
    display: block;
    position: relative;
  }

  .sam-session-dots {
    background-color: $purple3;
    width: 18px;
    height: 18px;
    padding: 8px 6px 12px 14px;
    margin: 10px;
    border-radius: 100px;
    border: 6px solid $purple3;
    position: absolute;
    @include std-transition;
    cursor: pointer;
  }

  .sam-session-dots:hover {
    border: 6px solid $purple5;
  }

  p {
    margin: 0;
    font-size: 18px;
    color: $purple1;
  }

  .sam-big-image {
  }

  .session-dots-1 {
    top: 280px;
    left: 5px;
  }

  .session-dots-2 {
    top: 283px;
    left: 248px;
  }

  .session-dots-3 {
    top: 283px;
    left: 498px;
  }

  .session-dots-4 {
    top: 130px;
    left: 149px;
  }

  .session-dots-5 {
    top: 125px;
    left: 325px;
  }

  .session-dots-6 {
    top: 129px;
    left: 679px;
  }

  .fair-image-holder {
    width: 525px;
    display: block;
    margin: 0 auto 50px;
  }

}