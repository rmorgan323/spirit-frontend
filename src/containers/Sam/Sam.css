.Sam .sam-graphic {
  width: 400px; }
