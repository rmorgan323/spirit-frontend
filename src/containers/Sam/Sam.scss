@import '../../variables.scss';

.Sam {

  .sam-image-holder {
    width: 668px;
    margin: 100px auto;
    display: block;
    position: relative;
  }

  .sam-session-dots {
    background-color: $purple3;
    width: 18px;
    height: 18px;
    padding: 8px 6px 12px 14px;
    margin: 10px;
    border-radius: 100px;
    border: 6px solid $purple3;
    position: absolute;
    @include std-transition;
    cursor: pointer;
  }

  .sam-session-dots:hover {
    border: 6px solid $purple5;
  }

  p {
    margin: 0;
    font-size: 18px;
    color: $purple1;
  }

  .sam-big-image {
  }

  .session-dots-1 {
    top: 137px;
    left: -14px;
  }

  .session-dots-2 {
    top: 140px;
    left: 206px;
  }

  .session-dots-3 {
    top: 141px;
    left: 439px;
  }

  .session-dots-4 {
    top: 0;
    left: 116px;
  }

  .session-dots-5 {
    top: -6px;
    left: 279px;
  }

  .session-dots-6 {
    top: 0;
    left: 605px;
  }

}