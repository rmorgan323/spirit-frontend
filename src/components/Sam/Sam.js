import React from 'react';
import { Link } from 'react-router-dom';
import './Sam.css';

const domains = [
  'modulation',
  'social',
  'postural',
  'sensory',
  'executive',
  ''
];

const Sam = () => {
  return (
    <div className="Sam">
      <div className="sam-image-holder">
        <img
          className="sam-big-image"
          src="/assets/sam-w-labels.svg"
          alt="sam-diagram"
        />
        {domains.map((domain, index) => {
          const domainNumber = index + 1;

          if (index === domains.length - 1) {
            return (
              <p key={domainNumber} className={`sam-session-dots session-dots-${domainNumber}`}>
                {domainNumber}
              </p>
            );
          }

          return (
            <Link key={domain} to={domain}>
              <p className={`sam-session-dots session-dots-${domainNumber}`}>
                {domainNumber}
              </p>
            </Link>
          );
        })}
      </div>
      <div className="fair-image-holder">
        <img
          className="fair-big-image"
          src="/assets/fair.svg"
          alt="fair-diagram"
        />
      </div>
    </div>
  );
};

export default Sam;
