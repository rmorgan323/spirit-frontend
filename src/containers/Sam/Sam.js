import React, { Component } from 'react';
import './Sam.css';

class Sam extends Component {
  constructor() {
    super();

    this.state = {
      imgSrc: `/assets/sam.png`
    }
  }

  handleMouseOver(num) {
    let array = [].slice.call(document.querySelectorAll(num))
    if (array.length) {
      array.forEach(val => val.classList.add('sam-hover'))
    }

    this.setState({
      imgSrc: `/assets/sam.png`
    })
  } 

  handleMouseOut(num) {
    let array = [].slice.call(document.querySelectorAll(num))
    if (array.length) {
      array.forEach(val => val.classList.remove('sam-hover'))
    }

    this.setState({
      imgSrc: `/assets/sam.png`
    })
  }

  render() {
    return (
      <div className="Sam">
        <img 
          className="sam-graphic" 
          src={this.state.imgSrc} 
          useMap="#sam_map"
          width="400"
          alt="sam-graphic"
        />
        <map name="sam_map">
          <area 
            className="sam-hover-1"
            shape="circle"
            coords="16,100,14"
            onMouseOver={() => this.handleMouseOver('.sam-1')}
            onMouseOut={() => this.handleMouseOut('.sam-1')}
            alt="sam-graphic"
          />
          <area 
            className="sam-hover-2"
            shape="circle"
            coords="285,102,14"
            onMouseOver={() => this.handleMouseOver('.sam-2')}
            onMouseOut={() => this.handleMouseOut('.sam-2')}
            alt="sam-graphic"
          />
          <area 
            className="sam-hover-3"
            shape="circle"
            coords="95,18,14"
            onMouseOver={() => this.handleMouseOver('.sam-3')}
            onMouseOut={() => this.handleMouseOut('.sam-3')}
            alt="sam-graphic"
          />
          <area 
            className="sam-hover-4"
            shape="circle"
            coords="148,102,14"
            onMouseOver={() => this.handleMouseOver('.sam-4')}
            onMouseOut={() => this.handleMouseOut('.sam-4')}
            alt="sam-graphic"
          />
          <area 
            className="sam-hover-5"
            shape="circle"
            coords="191,15,14"
            onMouseOver={() => this.handleMouseOver('.sam-5')}
            onMouseOut={() => this.handleMouseOut('.sam-5')}
            alt="sam-graphic"
          />
          <area 
            className="sam-hover-6"
            shape="circle"
            coords="384,18,14"
            onMouseOver={() => this.handleMouseOver('.sam-6')}
            onMouseOut={() => this.handleMouseOut('.sam-6')}
            alt="sam-graphic"
          />
        </map>
      </div>
    )
  }
}

export default Sam;















