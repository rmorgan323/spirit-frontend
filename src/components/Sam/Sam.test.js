import React from 'react';
import Sam from './Sam';
import { shallow } from 'enzyme';

describe('Sam tests', () => {
  it('Should match snapshot', () => {
    const renderedApp = shallow(<Sam />);

    expect(renderedApp).toMatchSnapshot();
  });
});
