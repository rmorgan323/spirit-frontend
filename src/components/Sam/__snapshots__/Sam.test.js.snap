// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sam tests Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sam />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sam-image-holder"
        >
          <img
            alt="sam-diagram"
            className="sam-big-image"
            src="/assets/sam-w-labels.svg"
          />
          <Link
            replace={false}
            to="modulation"
          >
            <p
              className="sam-session-dots session-dots-1"
            >
              1
            </p>
          </Link>
          <Link
            replace={false}
            to="social"
          >
            <p
              className="sam-session-dots session-dots-2"
            >
              2
            </p>
          </Link>
          <Link
            replace={false}
            to="postural"
          >
            <p
              className="sam-session-dots session-dots-3"
            >
              3
            </p>
          </Link>
          <Link
            replace={false}
            to="sensory"
          >
            <p
              className="sam-session-dots session-dots-4"
            >
              4
            </p>
          </Link>
          <Link
            replace={false}
            to="executive"
          >
            <p
              className="sam-session-dots session-dots-5"
            >
              5
            </p>
          </Link>
          <p
            className="sam-session-dots session-dots-6"
          >
            6
          </p>
        </div>,
        <div
          className="fair-image-holder"
        >
          <img
            alt="fair-diagram"
            className="fair-big-image"
            src="/assets/fair.svg"
          />
        </div>,
      ],
      "className": "Sam",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="sam-diagram"
              className="sam-big-image"
              src="/assets/sam-w-labels.svg"
            />,
            Array [
              <Link
                replace={false}
                to="modulation"
              >
                <p
                  className="sam-session-dots session-dots-1"
                >
                  1
                </p>
              </Link>,
              <Link
                replace={false}
                to="social"
              >
                <p
                  className="sam-session-dots session-dots-2"
                >
                  2
                </p>
              </Link>,
              <Link
                replace={false}
                to="postural"
              >
                <p
                  className="sam-session-dots session-dots-3"
                >
                  3
                </p>
              </Link>,
              <Link
                replace={false}
                to="sensory"
              >
                <p
                  className="sam-session-dots session-dots-4"
                >
                  4
                </p>
              </Link>,
              <Link
                replace={false}
                to="executive"
              >
                <p
                  className="sam-session-dots session-dots-5"
                >
                  5
                </p>
              </Link>,
              <p
                className="sam-session-dots session-dots-6"
              >
                6
              </p>,
            ],
          ],
          "className": "sam-image-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "sam-diagram",
              "className": "sam-big-image",
              "src": "/assets/sam-w-labels.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "modulation",
            "nodeType": "class",
            "props": Object {
              "children": <p
                className="sam-session-dots session-dots-1"
              >
                1
              </p>,
              "replace": false,
              "to": "modulation",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "sam-session-dots session-dots-1",
              },
              "ref": null,
              "rendered": 1,
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "social",
            "nodeType": "class",
            "props": Object {
              "children": <p
                className="sam-session-dots session-dots-2"
              >
                2
              </p>,
              "replace": false,
              "to": "social",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "sam-session-dots session-dots-2",
              },
              "ref": null,
              "rendered": 2,
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "postural",
            "nodeType": "class",
            "props": Object {
              "children": <p
                className="sam-session-dots session-dots-3"
              >
                3
              </p>,
              "replace": false,
              "to": "postural",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 3,
                "className": "sam-session-dots session-dots-3",
              },
              "ref": null,
              "rendered": 3,
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "sensory",
            "nodeType": "class",
            "props": Object {
              "children": <p
                className="sam-session-dots session-dots-4"
              >
                4
              </p>,
              "replace": false,
              "to": "sensory",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "sam-session-dots session-dots-4",
              },
              "ref": null,
              "rendered": 4,
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "executive",
            "nodeType": "class",
            "props": Object {
              "children": <p
                className="sam-session-dots session-dots-5"
              >
                5
              </p>,
              "replace": false,
              "to": "executive",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
                "className": "sam-session-dots session-dots-5",
              },
              "ref": null,
              "rendered": 5,
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "className": "sam-session-dots session-dots-6",
            },
            "ref": null,
            "rendered": 6,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="fair-diagram"
            className="fair-big-image"
            src="/assets/fair.svg"
          />,
          "className": "fair-image-holder",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "fair-diagram",
            "className": "fair-big-image",
            "src": "/assets/fair.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sam-image-holder"
          >
            <img
              alt="sam-diagram"
              className="sam-big-image"
              src="/assets/sam-w-labels.svg"
            />
            <Link
              replace={false}
              to="modulation"
            >
              <p
                className="sam-session-dots session-dots-1"
              >
                1
              </p>
            </Link>
            <Link
              replace={false}
              to="social"
            >
              <p
                className="sam-session-dots session-dots-2"
              >
                2
              </p>
            </Link>
            <Link
              replace={false}
              to="postural"
            >
              <p
                className="sam-session-dots session-dots-3"
              >
                3
              </p>
            </Link>
            <Link
              replace={false}
              to="sensory"
            >
              <p
                className="sam-session-dots session-dots-4"
              >
                4
              </p>
            </Link>
            <Link
              replace={false}
              to="executive"
            >
              <p
                className="sam-session-dots session-dots-5"
              >
                5
              </p>
            </Link>
            <p
              className="sam-session-dots session-dots-6"
            >
              6
            </p>
          </div>,
          <div
            className="fair-image-holder"
          >
            <img
              alt="fair-diagram"
              className="fair-big-image"
              src="/assets/fair.svg"
            />
          </div>,
        ],
        "className": "Sam",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="sam-diagram"
                className="sam-big-image"
                src="/assets/sam-w-labels.svg"
              />,
              Array [
                <Link
                  replace={false}
                  to="modulation"
                >
                  <p
                    className="sam-session-dots session-dots-1"
                  >
                    1
                  </p>
                </Link>,
                <Link
                  replace={false}
                  to="social"
                >
                  <p
                    className="sam-session-dots session-dots-2"
                  >
                    2
                  </p>
                </Link>,
                <Link
                  replace={false}
                  to="postural"
                >
                  <p
                    className="sam-session-dots session-dots-3"
                  >
                    3
                  </p>
                </Link>,
                <Link
                  replace={false}
                  to="sensory"
                >
                  <p
                    className="sam-session-dots session-dots-4"
                  >
                    4
                  </p>
                </Link>,
                <Link
                  replace={false}
                  to="executive"
                >
                  <p
                    className="sam-session-dots session-dots-5"
                  >
                    5
                  </p>
                </Link>,
                <p
                  className="sam-session-dots session-dots-6"
                >
                  6
                </p>,
              ],
            ],
            "className": "sam-image-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "sam-diagram",
                "className": "sam-big-image",
                "src": "/assets/sam-w-labels.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "modulation",
              "nodeType": "class",
              "props": Object {
                "children": <p
                  className="sam-session-dots session-dots-1"
                >
                  1
                </p>,
                "replace": false,
                "to": "modulation",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "className": "sam-session-dots session-dots-1",
                },
                "ref": null,
                "rendered": 1,
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "social",
              "nodeType": "class",
              "props": Object {
                "children": <p
                  className="sam-session-dots session-dots-2"
                >
                  2
                </p>,
                "replace": false,
                "to": "social",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2,
                  "className": "sam-session-dots session-dots-2",
                },
                "ref": null,
                "rendered": 2,
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "postural",
              "nodeType": "class",
              "props": Object {
                "children": <p
                  className="sam-session-dots session-dots-3"
                >
                  3
                </p>,
                "replace": false,
                "to": "postural",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 3,
                  "className": "sam-session-dots session-dots-3",
                },
                "ref": null,
                "rendered": 3,
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "sensory",
              "nodeType": "class",
              "props": Object {
                "children": <p
                  className="sam-session-dots session-dots-4"
                >
                  4
                </p>,
                "replace": false,
                "to": "sensory",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 4,
                  "className": "sam-session-dots session-dots-4",
                },
                "ref": null,
                "rendered": 4,
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "executive",
              "nodeType": "class",
              "props": Object {
                "children": <p
                  className="sam-session-dots session-dots-5"
                >
                  5
                </p>,
                "replace": false,
                "to": "executive",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 5,
                  "className": "sam-session-dots session-dots-5",
                },
                "ref": null,
                "rendered": 5,
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "host",
              "props": Object {
                "children": 6,
                "className": "sam-session-dots session-dots-6",
              },
              "ref": null,
              "rendered": 6,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="fair-diagram"
              className="fair-big-image"
              src="/assets/fair.svg"
            />,
            "className": "fair-image-holder",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "fair-diagram",
              "className": "fair-big-image",
              "src": "/assets/fair.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
